grails-app/controllers/org/pih/warehouse/InitializationInterceptor.groovy

```diff
@@ -12,45 +15,90 @@ import org.pih.warehouse.core.User
 import org.springframework.context.i18n.LocaleContextHolder
 import org.springframework.web.servlet.support.RequestContextUtils
 
+/**
+ * Hook-in for initializing a user session.
+ */
 class InitializationInterceptor {
 
+    @Value('${openboxes.locale.defaultLocale}')
+    String defaultLocale
+
+    @Value('${openboxes.locale.localizationModeLocale}')
+    String localizationModeLocale
+
+    @Value('${server.session.timeout}')
+    Integer sessionTimeoutInterval
+
     int order = HIGHEST_PRECEDENCE
 
-    public InitializationInterceptor() {
+    InitializationInterceptor() {
```

Comment:
I realized that we already had this class for building the user session so I switched to use it instead in an attempt to keep our code in one place.

I also cleaned up the code slightly by moving things into methods, adding comments, and accessing properties via @Value instead. None of the logic is changed.